src/main/java/org/prebid/server/bidder/taboola/TaboolaBidder.java

```diff
@@ -263,4 +269,24 @@ private BidType resolveBidType(String impId, List<Imp> imps) {
         }
         throw new PreBidException("Failed to find banner/native impression \"%s\"".formatted(impId));
     }
+
+    private static Bid resolveMacros(Bid bid) {
+        final BigDecimal price = bid.getPrice();
+        final String priceAsString = price != null ? price.toPlainString() : "0";
+
+        return bid.toBuilder()
+                .nurl(StringUtils.replace(bid.getNurl(), PRICE_MACRO, priceAsString))
+                .adm(StringUtils.replace(bid.getAdm(), PRICE_MACRO, priceAsString))
+                .build();
+    }
+
+    private static String extractGvlId(BidRequest bidRequest) {
+        return Optional.ofNullable(bidRequest)
+                .map(BidRequest::getExt)
+                .map(ExtRequest::getPrebid)
+                .map(ExtRequestPrebid::getServer)
+                .map(ExtRequestPrebidServer::getGvlId)
+                .map(Object::toString)
+                .orElse("");
```

Comment:
GvlId can be obtained directly via configuration properties. No need to use request.ext for that. Simply pass gvlId to adapter constructor.